Library/Homebrew/utils/github.rb

```diff
@@ -5,6 +5,7 @@
 require "utils/github/actions"
 require "utils/github/api"
 
+require "download_strategy"
```

Comment:
It's not the end of the world but I'm pretty sure that this will get included at startup now.

Reply:
Could move `download_artifact` elsewhere if it adds a non-negligible delay to startup.

Reply:
It already is a part of startup because of the API refactor - but that wasn't intentional and it maybe shouldn't so I'll have a look at that.

But with that said, I'm not really a fan of the tight coupling where this requires `download_strategy` and `download_strategy` requires `utils/github`.

Reply:
> though I realise we do that in a few other places as -W2 shouts at us for.

Note: if folks agree: enabling `-W1` and fixing issues (and then `-W2`) feels like a good `help wanted` issue I'd be happy to open.

Reply:
> But with that said, I'm not really a fan of the tight coupling where this requires `download_strategy` and `download_strategy` requires `utils/github` - though I realise we do that in a few other places as `-W2` shouts at us for.

Yeh, I also thought it was a little funky to have something in `utils/github` use a `DownloadStrategy`. Perhaps a new `utils/github/download` or something would be a better fit?

Reply:
> Note: if folks agree: enabling `-W1` and fixing issues

We already have done this for `-W1`, and is enabled by default now.

Haven't checked `-W2` in a while so there might be third-party stuff there, but there's definitely a lot of noise that is our fault and while not super important to fix, the warnings are there for a reason and are usually indicative of bad practices/smell.

Reply:
> > But with that said, I'm not really a fan of the tight coupling where this requires `download_strategy` and `download_strategy` requires `utils/github` - though I realise we do that in a few other places as `-W2` shouts at us for.
> 
> Yeh, I also thought it was a little funky to have something in `utils/github` use a `DownloadStrategy`. Perhaps a new `utils/github/download` or something would be a better fit?

Probably yeah. While we have these "mega" files like `download_strategy.rb` and `exceptions.rb`, it does introduce these type of coupling problems. I wouldn't mind `GitHubArtifactDownloadStrategy` to not live in `download_strategy.rb` and instead live in something like `utils/github/download` with the function that uses it.

Similarly, I very briefly drafted a custom download strategy for API download refactor stuff (before realising it was unnecessary) and for the short time that code existed, I placed it in `api/download.rb` instead.

I think it makes sense for `download_strategy.rb` to just be the base download strategies + the stuff used by formula and cask downloads while specific subclasses for particular cases can live elsewhere.

Reply:
Moved the artifact download code to a separate file.